packages/protocol-utils/src/signature_utils.ts

```diff
@@ -85,6 +85,44 @@ export async function eip712SignTypedDataWithProviderAsync(
     };
 }
 
+/**
+ * Sign a hash with the EthSign signature type using a compatible signature function
+ */
+export async function ethSignHashWithSignatureFnAsync(
+    hash: string,
+    signer: string,
+    signMessageAsync: (hash: string, signer: string) => Promise<string>,
+): Promise<Signature> {
+    if (typeof hash !== 'string') {
+        throw new Error(`Invalid message parameter for eth sign hash`);
+    }
```

Comment:
Are these checks necessary?

Reply:
With the current typing madness yeah, it would not stop you from passing `SignatureType.EIP712` and a string, or vice versa